src/shell/hooks/useLocalStorage.test.tsx

```diff
@@ -0,0 +1,193 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import React, { useCallback, useEffect, useState } from 'react';
+import { screen, within } from '@testing-library/react';
+import { exportForTest, useLocalStorage } from './useLocalStorage';
+import { setup } from '../../test/utils';
+
+describe('use local storage', () => {
+	const TestComponent = <T,>({
+		initialValue,
+		updatedValue,
+		localStorageKey = 'test-key'
+	}: {
+		initialValue: T;
+		updatedValue: React.SetStateAction<T>;
+		localStorageKey?: string;
+	}): JSX.Element => {
+		const [localStorageValue, setLocalStorageValue] = useLocalStorage(
+			localStorageKey,
+			initialValue
+		);
+		const localStorageStoreValue = exportForTest.useLocalStorageStore?.(
+			(state) => state.storage[localStorageKey]
+		);
+		const [localStorageUpdates, setLocalStorageUpdates] = useState<unknown[]>([]);
+		const [localStorageStoreUpdates, setLocalStorageStoreUpdates] = useState<unknown[]>([]);
+
+		useEffect(() => {
+			setLocalStorageUpdates((prevState) => [...prevState, localStorageValue]);
+		}, [localStorageValue]);
+
+		useEffect(() => {
+			setLocalStorageStoreUpdates((prevState) => [...prevState, localStorageStoreValue]);
+		}, [localStorageStoreValue]);
+
+		const setValueOnClick = useCallback((): void => {
+			setLocalStorageValue(updatedValue);
+		}, [setLocalStorageValue, updatedValue]);
+
+		return (
+			<>
+				<span>LocalStorageValue: {JSON.stringify(localStorageValue)}</span>
+				<span>LocalStorageValueUpdates: {JSON.stringify(localStorageUpdates)}</span>
+				<span>LocalStorageStoreValue: {JSON.stringify(localStorageStoreValue)}</span>
+				<span>LocalStorageStoreValueUpdates: {JSON.stringify(localStorageStoreUpdates)}</span>
+				<button onClick={setValueOnClick}>Update value</button>
+			</>
+		);
+	};
+
+	test('should show localStorage value and store value', () => {
+		const initial = 'initial';
+		const updated = 'updated';
+		const updatesLocalStorage = [initial];
+		const updatesLocalStorageStore = [undefined, initial];
+		setup(<TestComponent initialValue={initial} updatedValue={updated} />);
```

Comment:
Along with all the articles which uses react-hooks, I've found a couple of articles in which it is suggested to test hooks through simple test components. I found it useful for this case because it allowed me to test also the update of the data across different instances of the hook.

https://github.com/testing-library/react-hooks-testing-library
https://kentcdodds.com/blog/how-to-test-custom-react-hooks